common/core/web/keyboard-processor/src/text/kbdInterface.ts

```diff
@@ -948,20 +950,59 @@ namespace com.keyman.text {
       this.output(1, outputTarget, "");
     }
 
+    /**
+     * Function     processNewContextEvent
+     * Scope        Private
+     * @param       {Object}        outputTarget   The target receiving input
+     * @param       {Object}        keystroke      The input keystroke (with its properties) to be mapped by the keyboard.
+     * Description  Calls the keyboard's `begin newContext` group
+     * @returns     {number}        0 if no match is made, otherwise 1.
+     */
+    processNewContextEvent(outputTarget: OutputTarget, keystroke: KeyEvent): RuleBehavior {
+      if(!this.activeKeyboard) {
+        throw "No active keyboard for keystroke processing!";
+      }
+      return this.process(this.activeKeyboard.processNewContextEvent.bind(this.activeKeyboard), outputTarget, keystroke);
+    }
+
+    /**
+     * Function     processPostKeystroke
+     * Scope        Private
+     * @param       {Object}        outputTarget   The target receiving input
+     * @param       {Object}        keystroke      The input keystroke with relevant properties to be mapped by the keyboard.
+     * Description  Calls the keyboard's `begin postKeystroke` group
+     * @returns     {number}        0 if no match is made, otherwise 1.
+     */
+    processPostKeystroke(outputTarget: OutputTarget, keystroke: KeyEvent): RuleBehavior {
+      if(!this.activeKeyboard) {
+        throw "No active keyboard for keystroke processing!";
+      }
+      return this.process(this.activeKeyboard.processPostKeystroke.bind(this.activeKeyboard), outputTarget, keystroke);
+    }
+
     /**
      * Function     processKeystroke
      * Scope        Private
-     * @param       {Object}        element     The page element receiving input
+     * @param       {Object}        outputTarget   The target receiving input
      * @param       {Object}        keystroke   The input keystroke (with its properties) to be mapped by the keyboard.
      * Description  Encapsulates calls to keyboard input processing.
      * @returns     {number}        0 if no match is made, otherwise 1.
      */
     processKeystroke(outputTarget: OutputTarget, keystroke: KeyEvent): RuleBehavior {
+      if(!this.activeKeyboard) {
+        throw "No active keyboard for keystroke processing!";
+      }
+      return this.process(this.activeKeyboard.process.bind(this.activeKeyboard), outputTarget, keystroke);
+    }
+
+    private process(callee, outputTarget: OutputTarget, keystroke: KeyEvent): RuleBehavior {
```

Comment:
Is `callee` the same as `arguments.callee`? I see the latter has some limitations
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/arguments/callee

Reply:
No, `callee` is not related to `arguments.callee`. They just happen to have the same name... `callee` here is the function that will be called, see L1028.